subsys/net/ip/ipv6.c

```diff
@@ -93,17 +93,32 @@ static void ipv6_nbr_set_state(struct net_nbr *nbr,
 
 static inline bool net_is_solicited(struct net_pkt *pkt)
 {
-	return NET_ICMPV6_NA_HDR(pkt)->flags & NET_ICMPV6_NA_FLAG_SOLICITED;
+	struct net_icmpv6_na_hdr hdr, *na_hdr;
```

Comment:
That's nice, but increases stack usage for such funcs. As a further optimization, later we may want to read single field explicitly.